include/unifex/async_scope.hpp

```diff
@@ -632,21 +625,26 @@ class _attached_op<Sender, Receiver>::type final {
   using cleaning_receiver_t = cleaning_receiver<type, Receiver>;
   using nested_operation_t = connect_result_t<Sender, cleaning_receiver_t>;
 
+  struct stop_callback {
+    void operator()() noexcept { op_.request_stop(); }
+
+    type& op_;
+  };
+
 public:
   template <typename Receiver2>
   explicit type(Sender&& s, Receiver2&& r, async_scope* scope) noexcept(
       is_nothrow_connectable_v<Sender, cleaning_receiver_t>&&
-          std::is_nothrow_constructible_v<cleaning_receiver_t, type*, Receiver2>&&
+          std::is_nothrow_constructible_v<cleaning_receiver_t, type*>&&
               std::is_nothrow_constructible_v<Receiver, Receiver2>)
-    : scope_(reinterpret_cast<std::uintptr_t>(scope) & started_mask)
-    , receiverToken_(get_stop_token(r)) {
+    : scope_(reinterpret_cast<std::uintptr_t>(scope) & mask)
+    , receiver_((Receiver2 &&) r) {
     if (scope_ref()) {
```

Comment:
We should probably check `scope` here so we don't reload an atomic.
```suggestion
    if (scope) {
```